ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -992,17 +992,7 @@ private ServerErrorCode validateRequest(PartitionId partition, RequestOrResponse
       return ServerErrorCode.Bad_Request;
     }
     if (!skipPartitionAndDiskAvailableCheck) {
-      // 2. check if partition exists on this node and that the store for this partition has been started
-      if (storageManager.getStore(partition) == null) {
-        if (partitionsInCurrentNode.contains(partition)) {
-          metrics.diskUnavailableError.inc();
-          return ServerErrorCode.Disk_Unavailable;
-        } else {
-          metrics.partitionUnknownError.inc();
-          return ServerErrorCode.Partition_Unknown;
-        }
-      }
-      // 3. ensure the disk for the partition/replica is available
+      // 2. ensure the disk for the partition/replica is available
```

Comment:
What's the reason to swap order of 2 and 3? What is the thought here? (I am ok with this, but would like to why you swap the order).

Reply:
I am thinking `Disk_Unavailable` has a larger "failure domain" which means if the disk is down, all partitions on that disk would definitely be unavailable. In this case, I prefer to check disk availability first and then check partition availability.